src/hotspot/share/gc/parallel/psPromotionManager.inline.hpp

```diff
@@ -56,11 +56,13 @@ inline void PSPromotionManager::push_depth(ScannerTask task) {
 
 template <class T>
 inline void PSPromotionManager::claim_or_forward_depth(T* p) {
-  assert(should_scavenge(p, true), "revisiting object?");
```

Comment:
While working on [JDK-8310031](https://bugs.openjdk.org/browse/JDK-8310031) this assertion helped to find bugs quite a few times. It might be worth keeping it at line 62.